src/ORM/UniqueKey/UniqueKeyService.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+namespace SilverStripe\ORM\UniqueKey;
+
+use SilverStripe\Core\ClassInfo;
+use SilverStripe\Core\Injector\Injectable;
+use SilverStripe\ORM\DataObject;
+
+/**
+ * Class UniqueKeyService
+ *
+ * Generate a unique key for data object
+ *
+ * recommended use:
+ * - when you need unique key for caching purposes
+ * - when you need unique id on the front end (for example JavaScript needs to target specific element)
+ *
+ * @package SilverStripe\ORM\UniqueKey
+ */
+class UniqueKeyService implements UniqueKeyInterface
+{
+    use Injectable;
+
+    public function generateKey(DataObject $object, array $extraKeys = []): string
+    {
+        if (!$object->isInDB()) {
+            return '';
+        }
+
+        $class = ClassInfo::shortName($object);
```

Comment:
What happens if 2 dataobjects map to the same shortname?

Reply:
If you don't want to use FQCN you could use table name, maybe?

Reply:
The full FQDN is part of the key below this, the short name is just for legibility.

Reply:
As mentioned above - the hash is used to ensure uniqueness the rest is just additional information that helps debugging process.